### PR TITLE
Go through contributing docs

### PR DESCRIPTION
**Pull request by cicdw**
Our contribution docs read awkwardly to me so I went through them and touched them up. I also separated out documentation contributions from code contributions.

### CODE REVIEW

File: docs/snippets/fork.mdx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+All contributions to Prefect need to start on [a fork of the repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo).
+Once you have successfully forked [the Prefect repo](https://github.com/PrefectHQ/prefect), create a branch with an informative name:
+```
+git checkout -b fix-for-issue-NUM
+```
+After committing your changes to this branch, you can then open a pull request from your fork that we will review with you.
```
Comment (discdiver) [inline docs/snippets/fork.mdx:6]:
```suggestion
After committing your changes to this branch, you can then open a pull request from your fork that we will review.
```

File: docs/contribute/docs-contribute.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+import fork from '/snippets/fork.mdx'
+
+
+We use [Mintlify](https://mintlify.com/) to host and build the Prefect documentation.
+
+The main branch of the [prefecthq/prefect](https://github.com/PrefectHQ/prefect) GitHub repository is used to build the Prefect 3.0rc docs at [docs-3.prefect.io](https://docs-3.prefect.io).
+
+The 2.x docs are hosted at [docs.prefect.io](https://docs.prefect.io) and built from the 2.x branch of the repository.
```
Comment (discdiver) [inline docs/contribute/docs-contribute.mdx:13]:
```suggestion
The 2.x docs are hosted at [docs-2.prefect.io](https://docs-2.prefect.io) and built from the 2.x branch of the repository.
```

**Review by discdiver — APPROVED** [review]
Thank you! Minor edits.

File: docs/snippets/fork.mdx (hunk 1 above)
Comment (cicdw) [reply docs/snippets/fork.mdx:6]: The "with you" was intentional to highlight that review is a two-way conversation.

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**